### PR TITLE
Add module custom_tabulargseatochip

### PR DESCRIPTION
**Pull request by suzannejin**
Add the TABULAR_GSEA_TO_CHIP module from nf-core/differentialabundance to nf-core/modules repository.

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/custom/tabulartogseachip/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gawk:5.1.0' :
+        'biocontainers/gawk:5.1.0' }"
+
+    input:
+    path tabular
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/main.nf:11]:
```suggestion
    tuple val(meta), path(tabular)
```

File: modules/nf-core/custom/tabulartogseachip/meta.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - chip
+  - convert
+  - tabular
+tools:
+  - custom:
+      description: "Make a GSEA class file (.chip) from tabular inputs"
+      tool_dev_url: "https://github.com/nf-core/modules/blob/master/modules/nf-core/custom/tabulartogseachip/main.nf"
+      identifier: ""
+input:
+  - - tabular:
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/meta.yml:14]: Also meta here

File: modules/nf-core/custom/tabulartogseachip/meta.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: "custom_tabulartogseachip"
+description: Make a GSEA class file (.chip) from tabular inputs
+keywords:
+  - gsea
+  - chip
+  - convert
+  - tabular
+tools:
+  - custom:
+      description: "Make a GSEA class file (.chip) from tabular inputs"
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/meta.yml:10]:
```suggestion
      description: "Make a GSEA annotation file (.chip) from tabular inputs"
```

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Looking pretty good, thanks for the work, just a couple of minor things. Files are always supposed to have metas in the modules repo.

Not sure why this wasn't migrated before.

**Comment by pinin4fjords** [comment]
> @pinin4fjords Thank you for the review! Just one more question about the versioning... I copied it from the local version so I am not sure why VERSION=9.1?

Fair question, for which I don't remember the answer. But since we're using the gawk container, maybe you should just do what the gawk container does instead: https://github.com/nf-core/modules/blob/master/modules/nf-core/gawk/main.nf

File: modules/nf-core/custom/tabulartogseachip/main.nf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    output:
+    tuple val(meta), path("*.chip"), emit: chip
+    path "versions.yml"            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def VERSION = '9.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/main.nf:22]: You can strip this now

**Review by pinin4fjords — COMMENTED** [review]
Almost there would you mind adding a stub? Really easy for this one

**Comment by suzannejin** [comment]
@pinin4fjords I have just added the stub :)

File: modules/nf-core/custom/tabulartogseachip/main.nf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    tuple val(id)  , val(symbol)
+
+    output:
+    tuple val(meta), path("*.chip"), emit: chip
+    path "versions.yml"            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/main.nf:21]:
```suggestion
    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
```
Sorry, forgot that we should actually do the outputs this way

File: modules/nf-core/custom/tabulartogseachip/main.nf [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    function find_column_number {
+        file=\$1
+        column=\$2
+
+        head -n 1 \$file | tr '\\t' '\\n' | grep -n "^\${column}\$" | awk -F':' '{print \$1}'
+    }
+
+    id_col=\$(find_column_number $tabular $id)
+    symbol_col=\$(find_column_number $tabular $symbol)
+    outfile=\$(echo $tabular | sed 's/\\(.*\\)\\..*/\\1/').chip
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/main.nf:32]:
```suggestion
    outfile=${prefix}.chip
```

File: modules/nf-core/custom/tabulartogseachip/main.nf [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        gawk: \$(echo \$(gawk --version 2>&1) | sed 's/^.*GNU Awk //; s/, .*\$//')
+    END_VERSIONS
+    """
+
+    stub:
+    """
+    outfile=\$(echo $tabular | sed 's/\\(.*\\)\\..*/\\1/').chip
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/main.nf:46]:
```suggestion
    outfile=${prefix}.chip
```

**Review by pinin4fjords — COMMENTED** [review]
Sorry, just remembered we should also have a prefix

File: modules/nf-core/custom/tabulartogseachip/tests/main.nf.test.snap [hunk 8]
```diff
@@ -0,0 +1,42 @@
+{
+    "test tsv to chip": {
+        "content": [
+            [
+                [
+                    {
+                        "id": "test"
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/tests/main.nf.test.snap:7]:
```suggestion
                        "id": "SRP254919.salmon.merged.gene_counts.top1000cov"
```

File: modules/nf-core/custom/tabulartogseachip/tests/main.nf.test.snap [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+            "nextflow": "24.10.2"
+        },
+        "timestamp": "2024-12-16T18:05:22.341224384"
+    },
+    "test tsv to chip - stub": {
+        "content": [
+            [
+                [
+                    {
+                        "id": "test"
```
Comment (pinin4fjords) [inline modules/nf-core/custom/tabulartogseachip/tests/main.nf.test.snap:27]:
```suggestion
                        "id": "SRP254919.salmon.merged.gene_counts.top1000cov"
```

**Review by pinin4fjords — APPROVED** [review]
Looks good, thanks!

**Pull request merged**